Use new_observer_frame in Transform::look_at

JamesAngstrom opened the pull request:
Make the forward vector of and entity to point towards position (See #1134) when using Transform::look_at, as per the documentation. In particular, for a camera:

```rust
camera_transform.look_at(point, world_up);
```

should point the camera towards point, as in Unity's Transform.LookAt example:

https://docs.unity3d.com/ScriptReference/Transform.LookAt.html

I have an example that demonstrates this change here:

https://github.com/JamesAngstrom/look_at

you can switch between my branch and current amethyst master in Cargo.toml and see how the camera  behavior changes.

Before (camera rotates away from the sphere):
![before](https://user-images.githubusercontent.com/44684728/48505476-0d635f00-e83f-11e8-8277-317ed2384076.png)
After (camera points at sphere and rotates around it):
![after](https://user-images.githubusercontent.com/44684728/48505512-22d88900-e83f-11e8-9f24-131771c0cd26.png)

We could also have a separate function that has the new behavior with a different name, as suggested in the issue thread, but I think the above is likely going to be the expected behavior of look_at, and matches other game engines.

magnonellie commented:
I should probably note that what I meant with the new function name was to remove the old one entirely and just have the new one. That way people who used the function before will get compile errors since the behavior was incorrect in cgmath too. Then there's also the confusion between nalgebra and amethyst that occurs if it's kept this way which is probably the most important reason to rename it.

JamesAngstrom commented:
I changed the name of the function to face_towards as suggested. Since the name changed I also updated the changelog.

I also checked the example in the comment for the function and made sure it works. I also added a little extra using move_forward after face_towards because that might be more helpful to people that the existing rotation_between example.

Also I added some extra objects to my example to make sure it wasn't somehow a camera specific issue:
![rotatecubes](https://user-images.githubusercontent.com/44684728/48586210-3d862d00-e927-11e8-8be4-6de8a9407227.png)

magnonellie commented:
Why can't Linux be the only OS? Everything would have been so much easier that way 😣

bors retry
